conans/tools.py

```diff
@@ -23,14 +23,17 @@
 @contextmanager
 def pythonpath(conanfile):
     old_path = sys.path[:]
-    if conanfile.env.get("PYTHONPATH", None):
+    try:
         sys.path.extend(conanfile.env["PYTHONPATH"].split(os.pathsep))
+    except KeyError:
+        pass
     yield
     sys.path = old_path
 
 
 @contextmanager
-def environment_append(env_vars, keep_vars=["PATH", "PYTHONPATH"]):
+def environment_append(env_vars, keep_vars=None):
+    keep_vars = keep_vars or ["PATH", "PYTHONPATH"]
```

Comment:
Beware of keep_vers being an empty list. How do you specify you dont want to keep any var?